pkg/linode-bs/controllerserver.go

```diff
@@ -403,6 +403,11 @@ func (linodeCS *LinodeControllerServer) ListVolumes(ctx context.Context, req *cs
 	for _, vol := range volumes {
 		key := common.CreateLinodeVolumeKey(vol.ID, vol.Label)
 
+		var publishInfoVolumeName []string = make([]string, 0, 1)
+		if vol.LinodeID != nil {
+			publishInfoVolumeName = append(publishInfoVolumeName, fmt.Sprintf("%d", vol.LinodeID))
+		}
+
 		entries = append(entries, &csi.ListVolumesResponse_Entry{
```

Comment:
Is there a test covering the ListVolumes support?

Reply:
In short no there is not. 

In terms of unit testing `ListVolumes` there's the partially implemented fake linode client piece though given `linodeDriver.SetupLinodeDriver()` already accepts an interface we can mock out the LinodeAPI responses without too much hassle. I'd be happy to add a unit test as part of this PR. 

In terms of e2e tests, #140 is being worked on to fix those at the moment, but there's no direct testing of `ListVolumes` in those tests. Perhaps a discussion about using the [CSI sanity tests and Kubernetes E2E tests would make sense?](https://kubernetes.io/blog/2020/01/08/testing-of-csi-drivers/). I do think that's outside the scope of this PR however.

Reply:
Cool, unit test would be great. e2e tests can wait.

Reply:
Adding unit tests is going to be some work so I'm going to do it in a separate PR so this doesn't get overly busy.